test/jdk/com/sun/net/httpserver/bugs/B6431193.java

```diff
@@ -47,52 +47,45 @@ public static void read (InputStream i) throws IOException {
         i.close();
     }
 
-    /**
-     * @param args
-     */
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
         class MyHandler implements HttpHandler {
             public void handle(HttpExchange t) throws IOException {
-                InputStream is = t.getRequestBody();
-                read(is);
-                // .. read the request body
+                try (InputStream is = t.getRequestBody();
+                     OutputStream os = t.getResponseBody()) {
+                    read(is);
+                    // .. read the request body
                     String response = "This is the response";
-                t.sendResponseHeaders(200, response.length());
-                OutputStream os = t.getResponseBody();
-                os.write(response.getBytes());
-                os.close();
-                error = Thread.currentThread().isDaemon();
+                    t.sendResponseHeaders(200, response.length());
+                    os.write(response.getBytes());
+                    os.close();
+                    error = Thread.currentThread().isDaemon();
```

Comment:
Hello Darragh, since we are updating this test, perhaps we should rename that variable to be a bit more precise? Looking at the history of this test, it was introduced to verify that the HTTP request was handled on the server side in a non-daemon thread. So perhaps rename that field to `handlerIsDaemon`? Then, the place where we assert this value, perhaps change it to:

```java

if (handlerIsDaemon) {
    throw new RuntimeException ("request was handled by a daemon thread");
}
```